fundingmanager.go

```diff
@@ -557,43 +559,30 @@ func (f *fundingManager) start() error {
 				if err := ch.CloseChannel(closeInfo); err != nil {
 					fndgLog.Errorf("Failed closing channel "+
 						"%v: %v", ch.FundingOutpoint, err)
+					return
 				}
+				return
 
 			case <-f.quit:
 				// The fundingManager is shutting down, and will
 				// resume wait on startup.
-			case shortChanID, ok := <-confChan:
+				return
+
+			case shortChanID, ok = <-confChan:
 				if !ok {
 					fndgLog.Errorf("Waiting for funding" +
 						"confirmation failed")
 					return
 				}
+				// Fallthrough.
+			}
 
-				// The funding transaction has confirmed, so
-				// we'll attempt to retrieve the remote peer
-				// to complete the rest of the funding flow.
-				peerChan := make(chan lnpeer.Peer, 1)
-
-				var peerKey [33]byte
-				copy(peerKey[:], ch.IdentityPub.SerializeCompressed())
-
-				f.cfg.NotifyWhenOnline(peerKey, peerChan)
+			// Success, funding transaction was confirmed.
+			fndgLog.Debugf("ChannelID(%v) is now fully confirmed! "+
+				"(shortChanID=%v)", chanID, shortChanID)
 
-				var peer lnpeer.Peer
-				select {
-				case peer = <-peerChan:
-				case <-f.quit:
-					return
-				}
-				err := f.handleFundingConfirmation(
-					peer, ch, shortChanID,
-				)
-				if err != nil {
-					fndgLog.Errorf("Failed to handle "+
-						"funding confirmation: %v", err)
-					return
-				}
-			}
+			f.wg.Add(1)
+			go f.advanceFundingState(ch, nil)
```

Comment:
Nit: looks like there's no need to spawn this in another goroutine since we're already in one?

Reply:
It is just more natural, since the method decrements a waitgroup,